### ISSUE TITLE
Incompatibility with tasty 1.4

### ISSUE THREAD

**Issue by sjakobi** [phile314/tasty-silver#26]
`stack` reports:

```
tasty-silver    > configure
tasty-silver    > Configuring tasty-silver-3.1.15...
tasty-silver    > build
tasty-silver    > Preprocessing library for tasty-silver-3.1.15..
tasty-silver    > Building library for tasty-silver-3.1.15..
tasty-silver    > [1 of 6] Compiling Test.Tasty.Silver.Filter
tasty-silver    > 
tasty-silver    > /tmp/stack-b5982e9772b67286/tasty-silver-3.1.15/Test/Tasty/Silver/Filter.hs:26:1: warning: [-Wunused-imports]
tasty-silver    >     The import of ‘Data.Monoid’ is redundant
tasty-silver    >       except perhaps to import instances from ‘Data.Monoid’
tasty-silver    >     To import instances alone, use: import Data.Monoid()
tasty-silver    >    |
tasty-silver    > 26 | import Data.Monoid
tasty-silver    >    | ^^^^^^^^^^^^^^^^^^
tasty-silver    > 
tasty-silver    > /tmp/stack-b5982e9772b67286/tasty-silver-3.1.15/Test/Tasty/Silver/Filter.hs:114:9: warning: [-Wincomplete-patterns]
tasty-silver    >     Pattern match(es) are non-exhaustive
tasty-silver    >     In an equation for ‘filter'’: Patterns not matched: _ (After _ _ _)
tasty-silver    >     |
tasty-silver    > 114 |         filter' pth (SingleTest n t) = if prd (pth <//> n) then Just $ SingleTest n t else Nothing
tasty-silver    >     |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^...
tasty-silver    > [2 of 6] Compiling Test.Tasty.Silver.Interactive.Run
tasty-silver    > 
tasty-silver    > /tmp/stack-b5982e9772b67286/tasty-silver-3.1.15/Test/Tasty/Silver/Interactive/Run.hs:36:1: warning: [-Wincomplete-patterns]
tasty-silver    >     Pattern match(es) are non-exhaustive
tasty-silver    >     In an equation for ‘wrapRunTest'’:
tasty-silver    >         Patterns not matched: _ _ (After _ _ _)
tasty-silver    >    |
tasty-silver    > 36 | wrapRunTest' tp f (SingleTest n t) = SingleTest n (CustomTestExec t (f (tp <//> n) n))
tasty-silver    >    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^...
tasty-silver    > [3 of 6] Compiling Test.Tasty.Silver.Internal
tasty-silver    > 
tasty-silver    > /tmp/stack-b5982e9772b67286/tasty-silver-3.1.15/Test/Tasty/Silver/Internal.hs:13:1: warning: [-Wunused-imports]
tasty-silver    >     The import of ‘Data.Monoid’ is redundant
tasty-silver    >       except perhaps to import instances from ‘Data.Monoid’
tasty-silver    >     To import instances alone, use: import Data.Monoid()
tasty-silver    >    |
tasty-silver    > 13 | import Data.Monoid
tasty-silver    >    | ^^^^^^^^^^^^^^^^^^
tasty-silver    > [4 of 6] Compiling Test.Tasty.Silver.Interactive
tasty-silver    > 
tasty-silver    > /tmp/stack-b5982e9772b67286/tasty-silver-3.1.15/Test/Tasty/Silver/Interactive.hs:434:25: error:
tasty-silver    >     • Couldn't match type ‘Ap (Reader Level) TestOutput’
tasty-silver    >                      with ‘Ap (ReaderT Level Identity) TestOutput
tasty-silver    >                            -> Ap (ReaderT Level Identity) TestOutput’
tasty-silver    >       Expected type: OptionSet
tasty-silver    >                      -> TestName
tasty-silver    >                      -> Ap (ReaderT Level Identity) TestOutput
tasty-silver    >                      -> Ap (ReaderT Level Identity) TestOutput
tasty-silver    >         Actual type: TestName
tasty-silver    >                      -> Ap (Reader Level) TestOutput -> Ap (Reader Level) TestOutput
tasty-silver    >     • In the ‘foldGroup’ field of a record
tasty-silver    >       In the first argument of ‘foldTestTree’, namely
tasty-silver    >         ‘trivialFold
tasty-silver    >            {foldSingle = handleSingleTest, foldGroup = handleGroup}’
tasty-silver    >       In the second argument of ‘($)’, namely
tasty-silver    >         ‘foldTestTree
tasty-silver    >            trivialFold
tasty-silver    >              {foldSingle = handleSingleTest, foldGroup = handleGroup}
tasty-silver    >            opts
tasty-silver    >            tree’
tasty-silver    >     |
tasty-silver    > 434 |           , foldGroup = handleGroup
tasty-silver    >     |                         ^^^^^^^^^^^
tasty-silver    > 
tasty-silver    > /tmp/stack-b5982e9772b67286/tasty-silver-3.1.15/Test/Tasty/Silver/Interactive.hs:777:3: error:
tasty-silver    >     • Couldn't match type ‘Int’ with ‘Integer -> Maximum Int’
tasty-silver    >       Expected type: TestTree -> Integer -> Maximum Int
tasty-silver    >         Actual type: TestTree -> Int
tasty-silver    >     • Possible cause: ‘foldTestTree’ is applied to too many arguments
tasty-silver    >       In the second argument of ‘(.)’, namely
tasty-silver    >         ‘foldTestTree
tasty-silver    >            trivialFold
tasty-silver    >              {foldSingle = \ _ name _ level -> Maximum (length name + level),
tasty-silver    >               foldGroup = \ _ m -> m . (+ indentSize)}
tasty-silver    >            opts’
tasty-silver    >       In the expression:
tasty-silver    >         fromMonoid
tasty-silver    >           . foldTestTree
tasty-silver    >               trivialFold
tasty-silver    >                 {foldSingle = \ _ name _ level -> Maximum (length name + level),
tasty-silver    >                  foldGroup = \ _ m -> m . (+ indentSize)}
tasty-silver    >               opts
tasty-silver    >       In an equation for ‘computeAlignment’:
tasty-silver    >           computeAlignment opts
tasty-silver    >             = fromMonoid
tasty-silver    >                 . foldTestTree
tasty-silver    >                     trivialFold
tasty-silver    >                       {foldSingle = \ _ name _ level -> Maximum (length name + level),
tasty-silver    >                        foldGroup = \ _ m -> m . (+ indentSize)}
tasty-silver    >                     opts
tasty-silver    >             where
tasty-silver    >                 fromMonoid m
tasty-silver    >                   = case m 0 of
tasty-silver    >                       MinusInfinity -> 0
tasty-silver    >                       Maximum x -> x
tasty-silver    >     |
tasty-silver    > 777 |   foldTestTree
tasty-silver    >     |   ^^^^^^^^^^^^...
tasty-silver    > 
tasty-silver    > /tmp/stack-b5982e9772b67286/tasty-silver-3.1.15/Test/Tasty/Silver/Interactive.hs:779:22: error:
tasty-silver    >     • Couldn't match expected type ‘Int’
tasty-silver    >                   with actual type ‘Int -> Maximum Int’
tasty-silver    >     • In the ‘foldSingle’ field of a record
tasty-silver    >       In the first argument of ‘foldTestTree’, namely
tasty-silver    >         ‘trivialFold
tasty-silver    >            {foldSingle = \ _ name _ level -> Maximum (length name + level),
tasty-silver    >             foldGroup = \ _ m -> m . (+ indentSize)}’
tasty-silver    >       In the second argument of ‘(.)’, namely
tasty-silver    >         ‘foldTestTree
tasty-silver    >            trivialFold
tasty-silver    >              {foldSingle = \ _ name _ level -> Maximum (length name + level),
tasty-silver    >               foldGroup = \ _ m -> m . (+ indentSize)}
tasty-silver    >            opts’
tasty-silver    >     |
tasty-silver    > 779 |       { foldSingle = \_ name _ level -> Maximum (length name + level)
tasty-silver    >     |                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
tasty-silver    > 
tasty-silver    > /tmp/stack-b5982e9772b67286/tasty-silver-3.1.15/Test/Tasty/Silver/Interactive.hs:780:29: error:
tasty-silver    >     • Couldn't match type ‘[Char]’ with ‘Int -> Int’
tasty-silver    >       Expected type: Int -> Int
tasty-silver    >         Actual type: TestName
tasty-silver    >     • In the first argument of ‘(.)’, namely ‘m’
tasty-silver    >       In the expression: m . (+ indentSize)
tasty-silver    >       In the ‘foldGroup’ field of a record
tasty-silver    >     |
tasty-silver    > 780 |       , foldGroup = \_ m -> m . (+ indentSize)
tasty-silver    >     |                             ^
tasty-silver    > 
```

**Comment by phile314** [comment]
I pushed a fixed version on master. Any chance you could check if that works for you?

https://github.com/phile314/tasty-silver/commit/8834ac1943ba8a46add0c781bb320ebdb1cc63ac

**Comment by sjakobi** [comment]
> I pushed a fixed version on master. Any chance you could check if that works for you?

Cheers! The fix seems to work fine for me when I use it for the tests in https://github.com/dhall-lang/dhall-haskell/. :+1:

**Comment by phile314** [comment]
Released as v3.2:
http://hackage.haskell.org/package/tasty-silver-3.2
